src/components/dropdown-select.js

```diff
@@ -102,23 +117,16 @@ const DropdownSelect = ({
     let initialOptions = []
     let length = 0
 
-    if (data) {
-      if (simple) {
-        initialOptions = data
-        length = data.length
-      } else {
-        data.forEach((el) => {
-          el.items.forEach((item) => {
-            initialOptions.push(item)
-            length++
-          })
-        })
-      }
-    }
+    finalData?.forEach((el) => {
+      el.items.forEach((item) => {
+        initialOptions.push(item)
+        length++
+      })
+    })
 
     !limit && setSelectLimit(length)
     setOptions(initialOptions)
-  }, [data])
+  }, [finalData])
```

Comment:
leftover dependancy warning on `limit`

Reply:
yep, since I enabled `react-hooks` with this PR. There are a few other dependency warnings throughout certain components. In this PR, I only addressed lint errors that could be fixed automatically since introducing the new plugins.